hub/api/dataset.py

```diff
@@ -189,8 +192,31 @@ def load(
     @staticmethod
     @hub_reporter.record_call
     def delete(path: str, force: bool = False, large_ok: bool = False) -> None:
-        """Deletes a dataset"""
-        raise NotImplementedError
+        """Deletes a dataset at a given path.
+        This is an IRREVERSIBLE operation. Data once deleted can not be recovered.
+
+        Args:
+            path (str): The path to the dataset to be deleted.
+            force (bool): Delete data regardless of whether
+                it looks like a hub dataset. All data at the path will be removed.
+            large_ok (bool): Delete datasets larger than 1GB. Disabled by default.
+        """
+        base_storage = storage_provider_from_path(
+            path, creds={}, read_only=False, token=None
+        )
+
+        if force:
+            base_storage.clear()
```

Comment:
@mccrearyd @benchislett  Should we do `self.client.delete_dataset_entry(...)` for hub paths here? If not why do we need it for `ds.delete()`?